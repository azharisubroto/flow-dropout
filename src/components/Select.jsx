import PropTypes from 'prop-types'

const Select = ({ children, color, dark, width, height, fontSize, ...other }) => {
  const clr = color ? color : '#ececec'
  return (
    <select className={color} {...other}>
      {children}

      <style jsx>{`
        select {
          width: ${width ? `${width}px` : 'auto'};
          height: ${height ? `${height}px` : 'auto'};
          line-height: ${height ? `${height}px` : '1'};
          padding: ${!width ? '8px 13px' : '0'};
          font-size: ${fontSize ? fontSize : 12}px;
          display: flex;
          border-radius: 6px;
          background-color: ${clr};
          color: ${dark ? '#fff' : '#323a44'};
          border: 0;
          align-items: center;
          justify-content: center;
          cursor: pointer;
          transition: all 0.1s ease;
          outline-color: #d4eeea;

          &:hover {
            filter: brightness(104%);
          }
        }
      `}</style>
    </select>
  )
}

Select.propTypes = {
  children: PropTypes.node,
  color: PropTypes.string,
  dark: PropTypes.bool,
  fontSize: PropTypes.number,
  height: PropTypes.number,
  width: PropTypes.number
}

export default Select
