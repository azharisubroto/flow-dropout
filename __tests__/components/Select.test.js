import React from 'react'
import { shallow } from 'enzyme'
import Select from '@/components/Select'

describe('<Select />', () => {
  let wrapper, SelectComp

  beforeEach(() => {
    SelectComp = (
      <Select>
        <option>My Option</option>
      </Select>
    )

    wrapper = shallow(SelectComp)
  })

  it('should renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should contain <select> element', () => {
    expect(wrapper.find('select')).toBeTruthy()
  })

  it('should contain contain correct text', () => {
    expect(wrapper.find('select').html()).toEqual('<select><option>My Option</option></select>')
  })
})
