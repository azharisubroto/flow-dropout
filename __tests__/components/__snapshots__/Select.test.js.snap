// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> should renders correctly 1`] = `ShallowWrapper {}`;
